Sources/SwiftFusion/Core/Manifold.swift

```diff
@@ -0,0 +1,110 @@
+/// A coordinate in a differentiable manifold's global coordinate system.
+///
+/// Includes a coordinate chart [1] between `Self` and `R^n`, centered at `self`.
+///
+/// [1] https://en.wikipedia.org/wiki/Topological_manifold#Coordinate_charts
+public protocol ManifoldCoordinate: Differentiable {
+  /// The local coordinate system used in the chart.
+  ///
+  /// Isomorphic to `R^n`, where `n` is the dimension of the manifold.
+  associatedtype LocalCoordinate: AdditiveArithmetic & Differentiable & VectorProtocol
+    where LocalCoordinate.TangentVector == LocalCoordinate
```

Comment:
Good point.

It can't currently be part of `VectorProtocol` because `VectorProtocol` does not imply `Differentiable` and so `VectorProtocol` can't specify constraints related to `Differentiable` concepts. (`VectorProtocol` does not imply `Differentiable` because some user might want to do vector calculations without opting in to all the extra baggage of differentiability.)

I can think of some ways to make this better, so I'll make a note of this and we can consider it when rethinking `VectorProtocol`.